tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java

```diff
@@ -821,6 +823,43 @@ public void completeMultipartUpload() throws Exception {
     }
   }
 
+  @Test
+  public void bucketTags() throws Exception {
+    createBucketRestCall("test0");
+    createBucketRestCall("test1");
+
+    assertThat(getObjectTagRestCall("test0"), new StringContains("<TagSet/>"));
+    assertThat(getObjectTagRestCall("test1"), new StringContains("<TagSet/>"));
+
+    final HashMap<String, String> tag0 = new HashMap<>();
+    tag0.put("abc", "def");
+    tag0.put("ghi", "jkl");
```

Comment:
What are valid tags in s3? can we have empty keys or values? Can there be duplicate keys?

Also, is there a test to update the existing tags (and add new ones)?

Reply:
there can't be duplicate keys. (it wouldn't error but result is unexpected). Test to update the existing tags and add new ones isn't too useful here because what the api does isn't an update/remove. It just replaces the entire existing tags with new ones.